### ISSUE TITLE
LibreSSL 2.7.1 tests/timingsafe fails on Solaris 11

### ISSUE THREAD

**Issue by skrueger8** [libressl-portable/portable#413]
One test fails on Solaris 11:

OS: SunOS solaris 5.11 11.4.0.2.0 sun4v sparc sun4v
Compiler: cc: Studio 12.6 Sun C 5.15 SunOS_sparc 2017/05/30

```
==========================================
   libressl 2.7.1: tests/test-suite.log
==========================================

# TOTAL: 66
# PASS:  65
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: timingsafe
================

Assertion failed: (-cmp < 0) == (timingsafe_memcmp(buftwo, bufone, N) < 0), file timingsafe.c, line 48, function check
../test-driver: line 107: 23332: Abort(coredump)
FAIL timingsafe (exit status: 262)
```

**Comment by skrueger8** [comment]
Here's some poor man's debugging:

```
bufone[] = { 0x12 0xca 0x16 0xf5 0x76 0xc4 0x2e 0x67 };
buftwo[] = { 0x12 0xc5 0x40 0x13 0xb 0xc4 0x7f 0xb9 };
memcmp returned 5
timingsafe_memcmp(bufone, buftwo) returned -81
timingsafe_memcmp(buftwo, bufone) returned -5
Assertion failed: (cmp < 0) == (timingsafe_memcmp(bufone, buftwo, N) < 0), file timingsafe.c, line 67, function check
```

**Comment by skrueger8** [comment]
It's a bug in Solaris 11.4 Betas own timingsafe_memcmp() function.

Bug 27788630 - timingsafe_memcmp returns wrong results when later bytes match

I guess we could close this issue? The fix will hopefully end up in the Solaris 11.4 release.

**Comment by busterb** [comment]
Thanks!

I wasn't able to reproduce locally, I guess this would explain it... Perhaps Solaris' development team should incorporate this test into their own releases.
